hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -599,6 +599,9 @@ public Object invokeMethod(
           }
           LOG.error("Cannot get available namenode for {} {} error: {}",
               nsId, rpcAddress, ioe.getMessage());
+          if (this.namenodeResolver != null) {
+            this.namenodeResolver.rotateCache(nsId, namenode, shouldUseObserver);
```

Comment:
added log, Since this is a bug, most users must fix the bug by default, so I don’t think there is a need to add config here. I’m not sure there is a situation where users don’t need to fix the bug.